File: backend/app/schemas/customer.py
from pydantic import BaseModel


class PieSlice(BaseModel):
    name: str
    value: int
    percentage: float


class CustomerProfile(BaseModel):
    level_distribution: list[PieSlice]
    role_distribution: list[PieSlice]
    new_old_distribution: list[PieSlice]

File: frontend/src/api/customer.ts
import client, { type ApiResponse } from './client'

export interface PieSlice {
  name: string
  value: number
  percentage: number
}

export interface CustomerProfile {
  level_distribution: PieSlice[]
  role_distribution: PieSlice[]
  new_old_distribution: PieSlice[]
}

export const fetchCustomerProfile = () =>
  client.get<ApiResponse<CustomerProfile>>('/v1/customer/profile').then(r => r.data.data)
